mobile/library/swift/EngineBuilder.swift

```diff
@@ -612,6 +616,21 @@ open class EngineBuilder: NSObject {
     return self
   }
 
+  /// Adds a CDS layer to the configuration.
+  ///
+  /// - parameter cdsResourcesLocator: The layer name.
+  /// - parameter timeoutSeconds:      The timeout in seconds.
+  ///
+  /// - returns: This builder.
+  @discardableResult
+  public func addCDSLayer(cdsResourcesLocator: String = "", timeoutSeconds: UInt32 = 0) -> Self {
+    self.cdsResourcesLocator = cdsResourcesLocator
+    self.cdsTimeoutSeconds = timeoutSeconds
+    self.enableCds = true
+    return self
+  }
```

Comment:
```suggestion
  public func addCDSLayer(cdsResourcesLocator: String, timeoutSeconds: UInt32 = 0) -> Self {
    self.cdsResourcesLocator = cdsResourcesLocator
    self.cdsTimeoutSeconds = timeoutSeconds
    return self
  }
```

Reply:
> The idea being that it's legal to add a CDS config with no locator configured

Can you explain what the use case is for a CDS config with no locator?

Reply:
This is something @abeyad mentioned to me I think.  I'll leave it to him to verify or correct me on this.

Reply:
https://github.com/envoyproxy/envoy/blob/main/mobile/library/cc/engine_builder.h#L85 comment he wrote

Reply:
that's correct, cds resources locator is only for xdstp namespaces, for non-xdstp, nothing needs to be specified and a wildcard is inferred.